src/uct/ib/dc/base/dc_iface.c

```diff
@@ -358,10 +351,11 @@ int uct_dc_iface_is_reachable(const uct_iface_h tl_iface,
     uct_dc_iface_t UCS_V_UNUSED *iface = ucs_derived_of(tl_iface,
                                                         uct_dc_iface_t);
     uct_dc_iface_addr_t *addr = (uct_dc_iface_addr_t *)iface_addr;
+    uint8_t dc_ver_mask = UCT_DC_IFACE_ADDR_DC_V1 | UCT_DC_IFACE_ADDR_DC_V2;
 
     ucs_assert_always(iface_addr != NULL);
 
-    return (addr->flags & iface->version_flag) &&
+    return ((addr->flags & dc_ver_mask) == (iface->version_flag & dc_ver_mask)) &&
```

Comment:
can use `ucs_test_all_flags`

Reply:
need to test exact match
maybe make it shorter by doing only `((addr->flags & dc_ver_mask) == iface->version_flag)`